File: Python/basket.py
import exceptions


class Basket:
    def __init__(self):
        self._items = []

    def add_item(self, item):
        self._items.append(item)

    def calculate_result(self):
        if not self._items:
            raise exceptions.ListException()
        else:
            eatable_count = 0
            treatment_time = 0
            for item in self._items:
                if item.is_eatable():
                    eatable_count += 1
                else:
                    treatment_time += item.get_treatment()
            print(f"Eatable percent: {eatable_count / len(self._items) * 100}%")
            print(f"Treatment time: {treatment_time} minutes\n")

        self._items = []  # очистка кошику після розрахунку

File: C++/OOP Lab4/basket.cpp
#include "basket.h"
#include "exceptions.h"

Basket::~Basket() {
	for (auto item : items) {
		delete item;
	}
}

void Basket::addItem(Mushroom* item) {
	items.push_back(item);
}
void Basket::calculateResult() {
	if (items.empty()) {
		throw ListException();
	}

	int eatableCount = 0;
	int treatmentTime = 0;
	for (auto item : items) {
		if (item->isEatable()) {
			eatableCount++;
		}
		else {
			treatmentTime += item->getTreatment();
		}
	}
	std::cout << "Eatable percent: " << (int)(((float)eatableCount / (float)items.size()) * 100) << "%" << std::endl;
	std::cout << "Treatment time: " << treatmentTime << " minutes" << std::endl;
	system("pause");
	items.clear();
}